src/tir/transforms/compact_buffer_region.cc

```diff
@@ -303,28 +303,82 @@ class BufferAccessRegionCollector : public StmtExprVisitor {
   support::Arena arena_;
 };
 
+/*! \brief Collect storage alignment information from block annotations. */
+class StorageAlignCollector : public StmtVisitor {
+ public:
+  static std::unordered_map<Buffer, Array<Array<Integer>>, ObjectPtrHash, ObjectPtrEqual> Collect(
+      const PrimFunc& f) {
+    StorageAlignCollector collector;
+    collector(f->body);
+    return std::move(collector.storage_align_);
+  }
+
+ private:
+  void VisitStmt_(const BlockNode* op) final {
+    auto it = op->annotations.find(attr::buffer_dim_align);
+    if (it != op->annotations.end()) {
+      const auto& storage_align = Downcast<Array<Array<Array<Integer>>>>((*it).second);
```

Comment:
We don't need the const ref here

```suggestion
      auto storage_align = Downcast<Array<Array<Array<Integer>>>>((*it).second);
```